crates/biome_cli/src/commands/check.rs

```diff
@@ -102,7 +103,13 @@ pub(crate) fn check(
         organize_imports.enabled = organize_imports_enabled;
     }
 
-    fs_configuration.merge_with(configuration);
+    if let Some(mut configuration) = configuration {
+        if let Some(linter) = configuration.linter.as_mut() {
+            // Don't overwrite rules from the CLI configuration.
+            linter.rules = None;
+        }
+        fs_configuration.merge_with(configuration);
+    }
```

Comment:
Can you add a comment explaining why this is necessary? You gave some context in the PR, which should be described in the code, too.